src/dynamic_honey_badger/batch.rs

```diff
@@ -18,9 +20,30 @@ pub struct Batch<C, N> {
     pub(super) netinfo: Arc<NetworkInfo<N>>,
 }
 
+impl<C, N: NodeIdT> Epoched for Batch<C, N> {
+    type Epoch = Epoch;
+
+    /// Returns the **next** `DynamicHoneyBadger` epoch after the sequential epoch of the batch.
+    fn epoch(&self) -> Epoch {
+        let seqnum = self.seqnum;
+        let era = self.era;
+        if self.change == ChangeState::None {
+            Epoch(era, Some(seqnum - era + 1))
+        } else {
+            Epoch(seqnum + 1, Some(0))
+        }
+    }
+}
+
 impl<C, N: NodeIdT> Batch<C, N> {
-    pub fn epoch(&self) -> u64 {
-        self.epoch
+    /// Returns the linear epoch of this `DynamicHoneyBadger` batch.
+    pub fn seqnum(&self) -> u64 {
```

Comment:
Needs a better name.

Reply:
I can rename the `seqnum` field and method back to `epoch` and the `epoch` method to `next_epoch`.

Reply:
Maybe adding a `_num` or `_no` suffix to the methods which return a `u64` as opposed to an `Epoch` would help clarify things? 

Perhaps we should just leave it as is for now and create an issue to figure out a consistent naming scheme in a future PR.

Reply:
I also think we might need a more catchy name for this user-facing epoch of DHB. We already have two other, epochs that are used internally in HB and DHB. We are currently reusing the term "epoch" for the user-facing reduction of the real DHB epoch.